server/game/cards/challengeevent.js

```diff
@@ -0,0 +1,46 @@
+const DrawCard = require('../drawcard.js');
+
+class ChallengeEvent extends DrawCard {
+    // event card that can only be played during a challenge
+
+    constructor(owner, cardData, challengeType, participatingAs) {
+        // - challengeType (optional): make the event playable only on a
+        //   challenge of the given type
+        //
+        // - participatingAs (optional): make the event playable only when the
+        //   event controller is participating as either 'attacker' or
+        //   'defender'
+        //
+        super(owner, cardData);
+
+        this.challengeType = challengeType;
+        this.participatingAs = participatingAs;
+    }
+
+    canPlay(player, card) {
+        if(player !== this.controller || this !== card) {
+            return false;
+        }
+
+        var challenge = this.game.currentChallenge;
+        if(!challenge) {
+            return false;
+        }
+        if(this.challengeType
+           && challenge.challengeType !== this.challengeType) {
+            return false;
+        }
+        if((this.participatingAs === 'attacker'
```

Comment:
I find this block quite hard to read -  I can tell you've tried to wrap to 80 chars, but it's ok to go a bit longer these days, as much as 160 even.  Perhaps putting the two bracketed conditionals on their own line might make this clearer